fvp.mk

```diff
@@ -1,40 +1,29 @@
-BASH := $(shell which bash)
-ROOT ?= $(subst /build/,,$(shell pwd)/)
+-include common.mk
 
 ################################################################################
-# Paths to git projects and various binaries
+# Mandatory definition to use common.mk
 ################################################################################
-ARM_TF_PATH			?= $(ROOT)/arm-trusted-firmware
-
-EDK2_PATH 			?= $(ROOT)/edk2
-EDK2_BIN 			?= $(EDK2_PATH)/Build/ArmVExpress-FVP-AArch64/RELEASE_GCC49/FV/FVP_AARCH64_EFI.fd
-
-LINUX_PATH 			?= $(ROOT)/linux
-
-OPTEE_OS_PATH 			?= $(ROOT)/optee_os
-OPTEE_OS_BIN 			?= $(OPTEE_OS_PATH)/out/arm-plat-vexpress/core/tee.bin
-
-OPTEE_CLIENT_PATH 		?= $(ROOT)/optee_client
-OPTEE_CLIENT_EXPORT		?= $(OPTEE_CLIENT_PATH)/out/export
-OPTEE_LINUXDRIVER_PATH 		?= $(ROOT)/optee_linuxdriver
-
-OPTEE_TEST_PATH 		?= $(ROOT)/optee_test
-OPTEE_TEST_OUT_PATH 		?= $(ROOT)/out/optee_test
-
-GEN_ROOTFS_PATH 		?= $(ROOT)/gen_rootfs
-GEN_ROOTFS_FILELIST 		?= $(GEN_ROOTFS_PATH)/filelist-tee.txt
-
-FOUNDATION_PATH			?= $(ROOT)/Foundation_Platformpkg
+CROSS_COMPILE_NS_USER	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_NS_KERNEL	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_S_USER	?= "$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+CROSS_COMPILE_S_KERNEL	?= "$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+OPTEE_OS_BIN		?= $(OPTEE_OS_PATH)/out/arm-plat-vexpress/core/tee.bin
+OPTEE_OS_TA_DEV_KIT_DIR	?= $(OPTEE_OS_PATH)/out/arm-plat-vexpress/export-user_ta
 
 ################################################################################
-# defines, macros, configuration etc
+# Paths to git projects and various binaries
 ################################################################################
-define KERNEL_VERSION
-$(shell cd $(LINUX_PATH) && make kernelversion)
-endef
+ARM_TF_PATH		?= $(ROOT)/arm-trusted-firmware
+
+EDK2_PATH		?= $(ROOT)/edk2
+EDK2_BIN		?= $(EDK2_PATH)/Build/ArmVExpress-FVP-AArch64/RELEASE_GCC49/FV/FVP_AARCH64_EFI.fd
 
-CCACHE ?= $(shell which ccache) # Don't remove this comment (space is needed)
+GEN_ROOTFS_PATH		?= $(ROOT)/gen_rootfs
+GEN_ROOTFS_FILELIST	?= $(GEN_ROOTFS_PATH)/filelist-tee.txt
 
+FOUNDATION_PATH		?= $(ROOT)/Foundation_Platformpkg
+
+DEBUG = 1
```

Comment:
Instead of this, maybe set `DEBUG ?= 1` before `-include common.mk` to allow override with an environment variable? (and for consistency)